### PR TITLE
new tool rcorrector

### PR DESCRIPTION
**Pull request by bimbam23**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by bimbam23** [comment]
is someone able to help me with the error message? I thought I fix it by adding zlib as dependency.

**Comment by daler** [comment]
Hi @bimbam23 -  

No easy answer that I can see here, but a couple changes that may possibly help:

1. It looks like this tool is wget-ing the jellyfish source and compiling it (https://github.com/mourisl/Rcorrector/blob/master/Makefile#L13), but only if it doesn't exist. Adding a `- jellyfish >2` to the build and run deps should skip this step, and if the zlib error is happening on the jellyfish side, this may fix your error.

2. If you do end up needing zlib, it should be pinned (https://bioconda.github.io/linting.html#not-pinned). Pretty sure this should have been picked up by the linter but I'm not sure why and will look into that.

**Comment by bimbam23** [comment]
thanks for your help... It looks like there might be more dependencies... but zlib is still not working.

**Comment by daler** [comment]
Not sure why zlib isn't being found . . . @bioconda/lib any suggestions?

(By the way, you'll likely need jellyfish as a run dependency as well, once the zlib issue is resolved)

**Comment by dpryan79** [comment]
Add `export C_INCLUDE_PATH=$PREFIX/include` to the Makefile before `make`.

**Comment by dpryan79** [comment]
You'll want to remove the `{{CONDA_JELLYFISH}}` bit, that's not a predefined value for the template. Just use `>=2` or whatever is appropriate.

**Comment by bimbam23** [comment]
thanks Devon,
pinning everything does not help :-)

**Comment by bimbam23** [comment]
by the way, is the order important for the build and run?

**Comment by dpryan79** [comment]
Order doesn't seem to matter in any way.

**Comment by dpryan79** [comment]
Hmm, I guess my `export` trick failed.

**Comment by dpryan79** [comment]
You might also try setting CPLUS_INCLUDE_PATH or OBJC_INCLUDE_PATH. The one I listed is explicitly for C programs.

**Comment by bimbam23** [comment]
should I add all of them?

**Comment by dpryan79** [comment]
To see if any work, sure.

**Comment by dpryan79** [comment]
Looks like you might need to set `LD_LIBRARY_PATH=$PATH/lib` too.

File: recipes/rcorrector/build.sh [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 export C_INCLUDE_PATH=$PREFIX/include
 export OBJC_INCLUDE_PATH=$PREFIX/include
-export CPLUS_INCLUDE_PATH=$PREFIX/include
+export CPLUS_INCLUDE_PATH=$PREFIX/include 
+LD_LIBRARY_PATH=$PATH/lib
```
Comment (dpryan79) [inline recipes/rcorrector/build.sh]: You forgot `export`. You might want `export LD_LIBRARY_PATH=$PATH/lib:$LD_LIBRARY_PATH`. Thankfully conda will end up making a bunch of relative links out of things.

Comment (bimbam23) [reply recipes/rcorrector/build.sh]: see I know what I'm doing ;-)

**Comment by bimbam23** [comment]
now it looks good.

**Comment by epruesse** [comment]
Is there a reason you can't compile with LLVM on macos?

```
build:
  - gcc  # [lnux]
  - llvm  # [macos]
```

**Comment by bimbam23** [comment]
I don't know "llvm". Should I change to llvm to test it?

**Comment by bimbam23** [comment]
is there anything I can add/improve here?

**Comment by jerowe** [comment]
Looks good!

**Pull request merged**